server/game/core/ability/abilityTargets/CardTargetResolver.ts

```diff
@@ -87,11 +87,12 @@ export class CardTargetResolver extends TargetResolver<ICardTargetResolver<Abili
         // (unless there are dependent targets that might care about the targeting result)
         if (
             !this.dependentTarget &&
+            this.properties.mustChangeGameState !== GameStateChangeRequired.None &&
```

Comment:
If possible I'd like to find a solution that doesn't require explicitly setting `mustChangeGameState` in the target resolver properties like in DontGetCocky.ts, that property was mostly intended for internal engine use and is pretty technical so most people probably wouldn't understand how and when to set it. Could we get around this by just skipping this check if `this.getGameSystems()` is empty (i.e. no immediate effect in the target resolver)?